tests/FSharp.Compiler.ComponentTests/Interop/RequiredAndInitOnlyProperties.fs

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+namespace FSharp.Compiler.ComponentTests.Interop
+
+open Xunit
+open FSharp.Test.Compiler
+open FSharp.Test
+open System
+
+module ``Required and init-only properties`` =
+
+    let withCSharpLanguageVersion (ver: CSharpLanguageVersion) (cUnit: CompilationUnit) : CompilationUnit =
+        match cUnit with
+        | CS cs -> CS { cs with LangVersion = ver }
+        | _ -> failwith "Only supported in C#"
+
+
+    let csharpBaseClass = 
+        CSharp """
+    namespace RequiredAndInitOnlyProperties
+    {
+        public sealed class RAIO
+        {
+            public int GetSet { get; set; }
+            public int GetInit { get; init; }
+            public RAIO GetThis() => this;
+        }
+
+    }""" |> withCSharpLanguageVersion CSharpLanguageVersion.Preview |> withName "csLib"
+
+    let csharpRBaseClass = 
+        CSharp """
+    // Until we move to .NET7 runtime (or use experimental)
+    namespace System.Runtime.CompilerServices
+    {
+        [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
+        public sealed class RequiredMemberAttribute : Attribute { }
+        [AttributeUsage(AttributeTargets.All, AllowMultiple = true, Inherited = false)]
+        public sealed class CompilerFeatureRequiredAttribute : Attribute
+        {
+            public CompilerFeatureRequiredAttribute(string featureName)
+            {
+                FeatureName = featureName;
+            }
+            public string FeatureName { get; }
+            public bool IsOptional { get; init; }
+            public const string RefStructs = nameof(RefStructs);
+            public const string RequiredMembers = nameof(RequiredMembers);
+        }
+    }
+    
+    namespace RequiredAndInitOnlyProperties
+    {
+        public sealed class RAIO
+        {
+            public required int GetSet { get; set; }
+            public required int GetInit { get; init; }
+        }
+
```

Comment:
We should probably also have a test for when there is a constructor (which either does or doesn't set the required props, or some of them)

Reply:
This isn't supported scenario for F#.

Reply:
I will add tests to check that.